src/frontend/Deprecation_analysis.ml

```diff
@@ -0,0 +1,163 @@
+open Core_kernel
+open Ast
+open Middle
+
+let deprecated_functions =
+  String.Map.of_alist_exn
+    [ ("multiply_log", "lmultiply")
+    ; ("binomial_coefficient_log", "lchoose")
+    ; ("cov_exp_quad", "gp_cov_exp_quad")
+    ; ("integrate_ode", "ode_rk45")
+    ; ("integrate_ode_rk45", "ode_rk45")
+    ; ("integrate_ode_bdf", "ode_bdf")
+    ; ("integrate_ode_adams", "ode_adams") ]
```

Comment:
`integrate_ode_*` is deprecated but the corresponding `ode_*` is not a simple drop-in replacement. The old interface uses arrays and has tolerances at the end; the new interface uses vectors and has separate `ode_*_tol` function. If these are included in the list then the canonicalizer must not rely on `rename_function`.

Another case of a deprecated function without an easy substitution is `lkj_cov_log`. Stanc2 says
```
Info: the lkj_cov_log() sampling distribution is deprecated.  It will be removed in Stan 3.
Code LKJ covariance in terms of an lkj_corr() distribution on a correlation matrix and independent lognormals on the scales.
```
The canonicalizer ignored this because the required refactoring is too complicated.

Reply:
What would be a good way to handle `integrate_ode_*`, then? And do you think we should handle them differently in `emit_warnings` and `canonicalizer`?

Reply:
The canonicalizer cannot handle them at all. I think they need to be special-case in `emit_warnings` (along with `lkj_cov_log`)

Reply:
Sure. Which warning should be emitted, then? Something along the lines of
```
integrate_ode_x is deprecated and will be removed in the future. Use ode_x instead.
The new interface is slightly different, see:
https://mc-stan.org/users/documentation/case-studies/convert_odes.html
```
?

Reply:
That looks ok to me, not sure what to emit for lkj_cov_log, never used it to be honest.